R/pkg-nanoarrow.R

```diff
@@ -0,0 +1,23 @@
+
+as_nanoarrow_array_stream.DataFrame <- function(x, ..., schema = NULL) {
+  # Don't support the schema argument yet
+  stopifnot(is.null(schema))
+
+  stream = nanoarrow::nanoarrow_allocate_array_stream()
+  .pr$DataFrame$export_stream(x, nanoarrow::nanoarrow_pointer_addr_chr(stream))
+
+  stream
+}
+
+infer_nanoarrow_schema.DataFrame <- function(x, ...) {
+  as_nanoarrow_array_stream.DataFrame(x)$get_schema()
```

Comment:
@paleolimbot I know it's a late question, but isn't it a bad idea to create an array stream here and not release it?

Reply:
Thank you for the clear explanation!